app/controllers/api/mentor_discussion_posts_controller.rb

```diff
@@ -1,14 +1,14 @@
 module API
   class MentorDiscussionPostsController < BaseController
-    before_action :use_mentor_discussion
+    before_action :use_mentor_discussion, only: %i[index create]
 
     def index
       posts = @discussion.
         posts.
         joins(:iteration).
         where(iterations: { idx: params[:iteration_idx] })
 
-      render json: SerializeMentorDiscussionPosts.(posts)
+      render json: posts.map { |post| SerializeMentorDiscussionPost.(post, current_user) }
```

Comment:
Consider wrapping in an object, eg. `{ posts: [...] }`. It will make your life easier if you need metadata.